src/components/database.rs

```diff
@@ -57,11 +63,68 @@ impl DatabaseComponent {
         }
     }
 
-    pub fn get_repos(&self) -> &Option<DBRepositories> {
+    pub fn get_connection(db_connection: &Arc<Option<DBConnection>>) -> &DBConnection {
+        db_connection.as_ref().as_ref().unwrap()
+    }
+
+    pub async fn execute_query<'a>(
+        query: Query<'_, Postgres, PgArguments>,
+        transaction: Option<Transaction<'a, Postgres>>,
+        pool: &'a Pool<Postgres>,
+    ) -> (
+        Result<PgQueryResult, Error>,
+        Option<Transaction<'a, Postgres>>,
+    ) {
+        if let Some(mut executor) = transaction {
+            (query.execute(&mut executor).await, Some(executor))
+        } else {
+            (query.execute(pool).await, transaction)
+        }
+    }
+
+    pub async fn fetch_one<'a>(
```

Comment:
Same for this.

Reply:
refactored into receiving and returning an executor